src/eval.rs

```diff
@@ -1049,10 +1055,9 @@ where
                         }) => Closure { body: inner, env },
                         _ => unreachable!(),
                     }
-                }
-                // TODO: improve error message using some positions
-                else {
-                    return Err(EvalError::Other(String::from("empty metavalue"), pos));
+                } else {
+                    let label = meta.contracts.pop().unwrap().label;
```

Comment:
In some cases (e.g.: destructuring) the metavalue is a dummy metavalue. So it does not contains any value but  not any contracts neither. It may be an issue related to how is performed the destructuring now. Anyway, before merging, at least avoid a panic here is better.

Reply:
Good point. Maybe the error should take the label as an option then. That or we must enforce that any empty metavalue always come with a label, which does sound reasonable and is the case for anything that comes out of the parser. But it's still yet another invariant not verified by the compiler that we must maintain.

Reply:
Yes require a label for every metavalues song good. But I don't think we can do it staticaly because labels are attached to contracts which can be in 2 differents places. If we want, how to make it required? In which field (`types`/`contracts`)?

Reply:
It's not as simple than that: a metavalue with a non-empty `value` may very well have no label, and it's totally fine. What's not fine is a metavalue that is both empty and without label. Bu it's current panicking on destructuring, so let's handle the case where no label is defined before merging.

Reply:
Keeping in mind that, still, destructuring is not doing exactly the right thing right now.